src/Agent.Sdk/SecretMasking/OssSecretMasker.cs

```diff
@@ -81,11 +106,132 @@ public void Dispose()
     {
         _secretMasker?.Dispose();
         _secretMasker = null;
+        _telemetry = null;
     }
 
     public string MaskSecrets(string input)
     {
-        return _secretMasker.MaskSecrets(input);
+        _telemetry?.ProcessInput(input);
```

Comment:
>_telemetry [](http://example.com/codeflow?start=8&length=10)

OK, so looking at this, doesn't it indicate we've got something wrong (or an opportunity to do better) in the OSS masker interface?

i.e., although MaskSecrets has a per-detection mechanism, we don't have a good model for collecting general data on the discrete scan request. 

that seems like a problem. the only solution I can think of top-of-mind would be to add a second callback on mask secrets which is a sort of 'scanning target' general callback. eep. and should there be a 'scanning complete' callback (in that case, someone could bolt on their own profiling/timing collection).

Reply:
I think there's an opportunity to do something more general in the library, but I am not sure that this is exactly what we want. I think we could always do that later and then move over to it here.